account_move_template/models/account_move_template.py

```diff
@@ -0,0 +1,202 @@
+# Copyright 2015-2019 See manifest
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError, ValidationError
+from odoo.tools.safe_eval import safe_eval
+
+
+class AccountMoveTemplate(models.Model):
+    _name = "account.move.template"
+    _description = "Journal Entry Template"
+
+    name = fields.Char(required=True)
+    company_id = fields.Many2one(
+        "res.company",
+        string="Company",
+        required=True,
+        ondelete="cascade",
+        default=lambda self: self.env.company,
+    )
+    journal_id = fields.Many2one("account.journal", string="Journal", required=True)
+    ref = fields.Char(string="Reference", copy=False)
+    line_ids = fields.One2many(
+        "account.move.template.line", inverse_name="template_id", string="Lines"
+    )
+    active = fields.Boolean(default=True)
+
+    _sql_constraints = [
+        (
+            "name_company_unique",
+            "unique(name, company_id)",
+            "This name is already used by another template!",
+        )
+    ]
+
+    def copy(self, default=None):
+        self.ensure_one()
+        default = dict(default or {}, name=_("%s (copy)") % self.name)
+        return super().copy(default)
+
+    def eval_computed_line(self, line, sequence2amount):
+        safe_eval_dict = {}
+        for seq, amount in sequence2amount.items():
+            safe_eval_dict["L%d" % seq] = amount
+        try:
+            val = safe_eval(line.python_code, safe_eval_dict)
+            sequence2amount[line.sequence] = val
+        except ValueError as err:
```

Comment:
I use the `err` variable to bind to the raise with the from because without that a flake8 error would be generated.

"Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling."

Same on every case

Reply:
I understand, then leave it as it is.

@moylop260 hadn't we disabled raise ... from lint?